app/destination/page.js

```diff
@@ -63,75 +39,84 @@ export const Destinations = () => {
   ];
   const [selectedPlanets, setSelectedPlanets] = useState([]);
 
-  let isPlanetSelected = false;
+
   let numberOfPlanets = selectedPlanets.length;
 
-  const onAddOrRemovePlanet = (name, index) => {
+  const onAddOrRemovePlanet = (name) => {
+    setSelectedPlanets((prevSelectedPlanets) => {
+      const planet = planets.find((p) => p.name === name);
+
+      if (!prevSelectedPlanets.some((p) => p.name === name)) {
```

Comment:
A couple of things: 

1. It is better to remove the `!` and then swap the conditions for easier readability.
2. I would suggest to have two separate functions (and buttons if possible), one for adding and one for removing. That way the code structure becomes clearer :)